src/plugins/autoflow/AutoflowTabularPluginSpec.js

```diff
@@ -0,0 +1,320 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define([
+    './AutoflowTabularPlugin',
+    './AutoflowTabularConstants',
+    'zepto'
+], function (AutoflowTabularPlugin, AutoflowTabularConstants, $) {
+    describe("AutoflowTabularPlugin", function () {
+        var testType;
+        var testObject;
+        var mockmct;
+
+        beforeEach(function () {
+            testType = "some-type";
+            testObject = { type: testType };
+            mockmct = {
+                composition: jasmine.createSpyObj('composition', ['get']),
+                objectViews: jasmine.createSpyObj('views', ['addProvider']),
+                telemetry: jasmine.createSpyObj("telemetry", [
+                    'getMetadata',
+                    'getValueFormatter',
+                    'limitEvaluator',
+                    'request',
+                    'subscribe'
+                ])
+            };
+
+            var plugin = new AutoflowTabularPlugin({ type: testType });
+            plugin(mockmct);
+        });
+
+        it("installs a view provider", function () {
+            expect(mockmct.objectViews.addProvider).toHaveBeenCalled();
+        });
+
+        describe("installs a view provider which", function () {
+            var provider;
+
+            beforeEach(function () {
+                provider =
+                    mockmct.objectViews.addProvider.mostRecentCall.args[0];
+            });
+
+            it("applies its view to the type from options", function () {
+                expect(provider.canView(testObject)).toBe(true);
+            });
+
+            it("does not apply to other types", function () {
+                expect(provider.canView({ type: 'foo' })).toBe(false);
+            });
+
+            describe("provides a view which", function () {
+                var testKeys;
+                var testChildren;
+                var testContainer;
+                var testHistories;
+                var mockComposition;
+                var mockMetadata;
+                var mockEvaluator;
+                var mockUnsubscribes;
+                var callbacks;
+                var view;
+
+                function waitsForChange() {
+                    var oldHtml = testContainer.innerHTML;
+                    waitsFor(function () {
```

Comment:
Would it make sense to use `Vue.nextTick()` per [their testing docs](https://vuejs.org/v2/guide/unit-testing.html#Asserting-Asynchronous-Updates) for waiting for updates?

Reply:
Thanks for the link! I'll try that out

Reply:
That works well, but I opted for `window.requestAnimationFrame`, which accomplishes the same in practice, but allows the spec to remain totally Vue-agnostic